app/models/notification.rb

```diff
@@ -136,6 +136,20 @@ def update_notifications(notifiable, action = nil)
       Notifications::UpdateJob.perform_later(notifiable.id, notifiable.class.name, action)
     end
 
+    def fast_destroy_old_notifications(destroy_before_timestamp = 4.months.ago)
+      notification_sql = <<-SQL
+        DELETE FROM notifications
+        WHERE notifications.id IN (
+          SELECT notifications.id
+          FROM notifications
+          WHERE created_at < '#{destroy_before_timestamp}'
```

Comment:
this makes me a little bit uncomfortable, because we're joining strings to build a SQL query. What if we pass the params to the bulk class and let it use them with `binds` in the other class:

```ruby
connection.exec_delete(sql, "DELETE NOTIFICATIONS", [[nil, destroy_before_timestamp]])
```

see https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-exec_delete

Reply:
I feel like this is easier to read this way and I like that we have full control over the statement and the limit we put on it

Reply:
the issue is that string concatenation can lead to involountary SQL injection. It's highly unlikely but it's a thing that I thought about while reviewing. Also it might help validating 
parameters if we would need to :)